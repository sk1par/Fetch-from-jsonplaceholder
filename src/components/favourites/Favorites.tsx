import { useSelector } from 'react-redux';
import IUser from '../../models/user.interface';
import { addToFavorite } from '../../selectors/selectors';
import FavoriteUser from '../favorite-user/FavoriteUser';

const Favorites = () => {
    const users: IUser[] = useSelector(addToFavorite);

    return (
        <div className="container mt-4" data-auto-id='component-container'>
            <div className="row">
                {users?.map((user: IUser) => (
                    <FavoriteUser user={user} key={user.id} data-auto-id='user-div-row' />
                ))}
            </div>
        </div>
    )
}

export default Favorites
