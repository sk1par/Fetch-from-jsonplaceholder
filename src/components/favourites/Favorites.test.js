import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Favorites from './Favorites';
import { useSelector } from 'react-redux';

Enzyme.configure({ adapter: new Adapter() });

jest.mock('react-redux', () => ({
    useDispatch: () => (fn) => () => fn(),
    useSelector: jest.fn(() => []),
}));

const getInitialObject = () => [{
    email: 'test@abv.bg',
    id: 1,
    name: 'test',
    phone: '123132',
    username: 'skipar',
    website: 'abv.bg'
}];

it('should render component correctly', () => {
    const wrapper = shallow(<Favorites />);
    const componentContainer = wrapper.find('[data-auto-id="component-container"]');
    expect(componentContainer).toHaveLength(1);
    const items = wrapper.find('[data-auto-id="user-div-row"]');
    expect(items).toHaveLength(0);
    expect(wrapper).toMatchSnapshot();
});

it('should render list of two mock favorite users', () => {
    useSelector.mockReturnValueOnce([getInitialObject(), getInitialObject()]);
    const wrapper = shallow(<Favorites />);
    const items = wrapper.find('[data-auto-id="user-div-row"]');
    expect(items).toHaveLength(2);
    expect(wrapper).toMatchSnapshot();
});
