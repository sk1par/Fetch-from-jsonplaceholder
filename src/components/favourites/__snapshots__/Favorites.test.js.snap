// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component correctly 1`] = `ShallowWrapper {}`;

exports[`should render list of two mock favorite users 1`] = `ShallowWrapper {}`;
